### PR TITLE
Add NetBSD target to build scripts

### PR DESCRIPTION
**Pull request by krytarowski**
It pushes this target towards independentency.

### CODE REVIEW

**Comment by krytarowski** [comment]
@stephentoub @ellismg PTAL

**Comment by krytarowski** [comment]
Does this commit make sense?

**Comment by krytarowski** [comment]
I want to be building assemblies for NetBSD as soon as possible and be adding missing dependencies.

**Comment by stephentoub** [comment]
> I want to be building assemblies for NetBSD as soon as possible and be adding missing dependencies.

For all but a handful of assemblies (System.Diagnostics.Process, System.IO.FileSystem.Watcher, System.Runtime.InteropServices.RuntimeInformation, System.IO.FileSystem.DriveInfo, and System.Net.NetworkInformation, I believe), there should be zero difference between those built for Linux/OSX and those built for NetBSD... it's a failing of our current set up that we even have these distinctions in the .csproj files, as the actual built code doesn't distinguish at all, and I'd like to see if we can rectify that rather than further expanding out the number of platforms listed.  In reality, most of these projects really only need a Windows and Unix configuration, not one for each of Windows, Linux, OSX, FreeBSD, NetBSD, and so on.  In the meantime, with the exception of those few assemblies mentioned, you can just build with OSGroup=Linux.  And for these few assemblies, there's a lot more to be done than just changing a few lines in the .csproj files.

**Comment by stephentoub** [comment]
@weshaggard, @ericstj, what would you recommend we do to stop having this continued expansion of builds for the majority of our assemblies that have a Unix build that doesn't further differentiate between Linux, OSX, etc.?  Do we have the ability to just say "Unix_Debug", "Unix_Release", etc., and have that flavor built when you specify an OSGroup of Linux, OSX, NetBSD, etc.?

**Comment by krytarowski** [comment]
I want to fix these OS specific ones. For example in the filesystem watcher I have an idea to reuse Linux-compat inotify interface and so forth. (Yes, I know that I can work on a pile of local patches, but I prefer to work closely to upstream).

If you think there is a place to narrow down these tests to Unix/Windows and distinguish OS-flavors for a few of them please help enable it.

Thank you!

**Comment by krytarowski** [comment]
I will admit that my patch is for sure incomplete and it will be likely a continued burden to maintain.

**Comment by stephentoub** [comment]
> I want to fix these OS specific ones

Great.  Please focus on one of those at a time then, rather than changing all of these files that I'm hoping we won't actually need to change like this.  I'd be happy to see a single PR that enabled building System.IO.FileSystem.Watcher with support for NetBSD, including whatever inotify changes are necessary.

File: dir.props [hunk 1]
```diff
@@ -276,6 +277,13 @@
         <TestNugetRuntimeId>ubuntu.14.04-x64</TestNugetRuntimeId>
       </PropertyGroup>
     </When>
+    <When Condition="'$(OSGroup)'=='NetBSD'">
+      <PropertyGroup>
+        <TargetsUnix>true</TargetsUnix>
+        <TargetsNetBSD>true</TargetsNetBSD>
+        <TestNugetRuntimeId>ubuntu.14.04-x64</TestNugetRuntimeId>
+      </PropertyGroup>
+    </When>
```
Comment (stephentoub) [inline dir.props]: I'd recommend you simplify this PR down to basically the changes in this file.

**Comment by krytarowski** [comment]
Right now I'm researching signal handlers of SIGSEGV - this is platform (OS+hardware) dependent and the behavior is different on NetBSD and other supported Unices (Linux + MacOSX). Once squashed I will reiterate over bigger issues -- likely hangs of several tests.

File: dir.props (hunk 1 above)
Comment (krytarowski) [inline dir.props]: OK!

**Comment by krytarowski** [comment]
Patch has been reduced to the single file.

I'm looking forward to see changes allowing dedicated NetBSD target!

**Comment by stephentoub** [comment]
LGTM

> I'm looking forward to see changes allowing dedicated NetBSD target!

This change does that.  You should be able to build any project with /p:OSGroup=NetBSD after this (except for the few that depend on TargetsUnknownUnix, which you're modifying in your changes to say that NetBSD is no longer "unknown" and thus the stub files associated with TargetsUnknownUnix in projects like System.Diagnostics.Process won't be included in the build.)

**Comment by krytarowski** [comment]
> This change does that. You should be able to build any project with /p:OSGroup=NetBSD after this (except for the few that depend on TargetsUnknownUnix, which you're modifying in your changes to say that NetBSD is no longer "unknown" and thus the stub files associated with TargetsUnknownUnix in projects like System.Diagnostics.Process won't be included in the build.)

Thank you, it's clear now.

**Comment by krytarowski** [comment]
Thank you!

**Comment by weshaggard** [comment]
@stephentoub I have issue https://github.com/dotnet/corefx/issues/5420 assigned to me to try and add support for "Unix" but I just haven't had the cycles to tackle it yet. If someone is interested in trying to tackle it I'm happy to provide guidance and reviews.

**Comment by stephentoub** [comment]
Thanks. Glad there's an issue open for it.

**Pull request merged**